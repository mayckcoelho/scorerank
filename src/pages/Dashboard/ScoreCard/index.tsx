import React, { useState } from 'react'
import Collapsible from 'react-native-collapsible';
import Icon from "react-native-vector-icons/FontAwesome";
Icon.loadFont();

import Card from "../../../components/Card";
import { useAuth } from '../../../contexts/auth';
import ScoreColors from "../ScoreColors.json";
import {
    ScoreText,
    ScoreDescription,
    Separator,
    SeeMore,
    SeeMoreText,
    ScoreTemplateText,
    ScoreTemplateItem,
    Gradient
} from "./styles";

const Dashboard: React.FC = () => {
    const [isCollapsed, setIsCollapsed] = useState(true)
    const { user } = useAuth();

    let colors;
    if (user) {
        if (user.score < 30)
            colors = ScoreColors.find(c => c.step === 30);
        else if (user.score < 60)
            colors = ScoreColors.find(c => c.step === 60);
        else
            colors = ScoreColors.find(c => c.step === 100);
    }

    return (
        <Card>
            <ScoreText>{user?.score}</ScoreText>
            <ScoreDescription>Sua pontuação é {colors?.description}</ScoreDescription>
            <Separator />
            <SeeMore onPress={() => setIsCollapsed(!isCollapsed)}>
                <SeeMoreText>Saiba mais</SeeMoreText>
                <Icon name={isCollapsed ? "chevron-down" : "chevron-up"} size={16} color="#6a1e55" />
            </SeeMore>
            <Collapsible style={{ flexDirection: "row" }} collapsed={isCollapsed}>
                <ScoreTemplateItem>
                    <ScoreTemplateText>0 a 30</ScoreTemplateText>
                    <Gradient start={{ x: 0, y: 0 }} end={{ x: 1, y: 1 }} colors={ScoreColors[0].colors.gradient as React.ReactText[]} />
                </ScoreTemplateItem>
                <ScoreTemplateItem>
                    <ScoreTemplateText>31 a 60</ScoreTemplateText>
                    <Gradient start={{ x: 1, y: 1 }} end={{ x: 0, y: 0 }} colors={ScoreColors[1].colors.gradient as React.ReactText[]} />
                </ScoreTemplateItem>
                <ScoreTemplateItem>
                    <ScoreTemplateText>61 a 100</ScoreTemplateText>
                    <Gradient start={{ x: 1, y: 1 }} end={{ x: 0, y: 0 }} colors={ScoreColors[2].colors.gradient as React.ReactText[]} />
                </ScoreTemplateItem>
            </Collapsible>
        </Card>
    )
};

export default Dashboard;