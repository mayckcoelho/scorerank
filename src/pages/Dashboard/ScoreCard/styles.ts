import styled from "styled-components/native";
import { Animated } from "react-native";
import LinearGradient from 'react-native-linear-gradient';

export const ScoreText = styled.Text`
    margin-bottom: 20px;
    font-size: 50px;
    font-weight: bold;
    color: #676767;
    align-self: center;
`

export const ScoreDescription = styled.Text`
    font-size: 20px;
    color: #676767;
    align-self: center;
`
export const Separator = styled.View`
    background-color: #676767;
    width: 100%;
    height: 1px;
    margin: 20px 0;
`

export const SeeMore = styled.TouchableOpacity`
    flex-direction: row;
    justify-content: center;
    align-items: center;
`

export const SeeMoreText = styled.Text`
    font-size: 18px;
    color: #6a1e55;
    align-self: center;
    font-weight: bold;
    margin-right: 5px;
`

export const ScoreTemplateText = styled.Text`
    font-size: 16px;
    color: #676767;
    align-self: center;
`

export const ScoreTemplateItem = styled.View`
    flex:1;
    margin: 40px 1px 0;
`

export const Gradient = styled(LinearGradient)`
    margin-top: 20px;
    height: 13px;
`