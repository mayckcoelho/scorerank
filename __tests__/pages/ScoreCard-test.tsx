import React from 'react';
import ScoreCard from '../../src/pages/Dashboard/ScoreCard';
import renderer from 'react-test-renderer';

describe('ScoreCard should render correctly', () => {
    it('match snapshot', () => {
        const scorecard = renderer
            .create(<ScoreCard />)
            .toJSON();
        expect(scorecard).toMatchSnapshot();
    });
})