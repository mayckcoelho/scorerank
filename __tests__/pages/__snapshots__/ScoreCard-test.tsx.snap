// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreCard should render correctly match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 6,
        "marginBottom": 20,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 20,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "shadowColor": "rgba(0, 0, 0, .3)",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 8,
        "width": "95%",
      },
      Object {
        "backgroundColor": "#fff",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#676767",
          "fontSize": 50,
          "fontWeight": "bold",
          "marginBottom": 20,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#676767",
          "fontSize": 20,
        },
      ]
    }
  >
    Sua pontuação é 
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#676767",
          "height": 1,
          "marginBottom": 20,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 20,
          "width": "100%",
        },
      ]
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#6a1e55",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginRight": 5,
          },
        ]
      }
    >
      Saiba mais
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#6a1e55",
            "fontSize": 16,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    pointerEvents="none"
    style={
      Object {
        "height": 0,
        "overflow": "hidden",
      }
    }
  >
    <View
      forwardedRef={[Function]}
      onLayout={[Function]}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 0,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 40,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#676767",
                "fontSize": 16,
              },
            ]
          }
        >
          0 a 30
        </Text>
        <BVLinearGradient
          colors={
            Array [
              4291455544,
              4292850751,
            ]
          }
          endPoint={
            Object {
              "x": 1,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          style={
            Array [
              Object {
                "height": 13,
                "marginTop": 20,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 0,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 40,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#676767",
                "fontSize": 16,
              },
            ]
          }
        >
          31 a 60
        </Text>
        <BVLinearGradient
          colors={
            Array [
              4286627517,
              4292064078,
            ]
          }
          endPoint={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 1,
              "y": 1,
            }
          }
          style={
            Array [
              Object {
                "height": 13,
                "marginTop": 20,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 0,
              "marginLeft": 1,
              "marginRight": 1,
              "marginTop": 40,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#676767",
                "fontSize": 16,
              },
            ]
          }
        >
          61 a 100
        </Text>
        <BVLinearGradient
          colors={
            Array [
              4280308829,
              4285906872,
            ]
          }
          endPoint={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 1,
              "y": 1,
            }
          }
          style={
            Array [
              Object {
                "height": 13,
                "marginTop": 20,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
